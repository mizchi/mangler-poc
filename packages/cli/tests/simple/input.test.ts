import fs from "node:fs";
import path from "node:path";
import { beforeAll, expect, test } from "vitest";
import { bundle, execBuild } from "../_testUtils";

const cwd = path.dirname(new URL(import.meta.url).pathname);

let analyzed: any;
beforeAll(async () => {
  await execBuild(cwd);
  analyzed = JSON.parse(fs.readFileSync(path.join(cwd, "analyzed.json"), "utf-8"));
});

test("compile with analyze", async () => {
  const fileName = path.join(cwd, "input.ts");
  const out = await bundle(fileName, {
    mangle: {
      properties: {
        regex: /.*/,
        reserved: analyzed.reserved,
      },
    },
  });
  expect(analyzed.reserved).toEqual(["0", "1", "100", "A", "B", "C", "a", "b", "bar", "f", "g"]);
  expect(out).includes("bar");
  expect(out).includes(".filter");
  expect(out).includes("console.log");
  expect(out).not.includes("xxxxx");
  expect(out).not.includes("internal");
  expect(out).not.includes("pf", "z");
  expect(out).toMatchSnapshot();
});

test("break without terser builtins", async () => {
  const fileName = path.join(cwd, "input.ts");
  const out = await bundle(fileName, {
    mangle: {
      properties: {
        builtins: true,
        regex: /.*/,
        reserved: analyzed.reserved,
      },
    },
  });
  expect(out).not.includes(".filter");
});
