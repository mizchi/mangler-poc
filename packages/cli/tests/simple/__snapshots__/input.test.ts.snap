// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compile with analyze 1`] = `
"const n={bar:1};function o(){return[].filter((n=>n))}class c{a=1;b;f(){}g(){}o(){}z=null}var r;!function(n){n[n.A=0]=\\"A\\",n[n.B=1]=\\"B\\",n[n.C=100]=\\"C\\"}(r||(r={}));console.log(1);export{r as MyEnum,c as X,o as f,n as foo};
"
`;
